File: app/src/main/java/com/coopra/nebulus/TokenHandler.java
package com.coopra.nebulus;

import android.content.Context;
import android.content.SharedPreferences;

public class TokenHandler {
    public static void saveToken(Context context, String token) {
        SharedPreferences sharedPref = context.getSharedPreferences(context.getString(R.string.preference_file_key), Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPref.edit();
        editor.putString(context.getString(R.string.saved_token), token);
        editor.commit();
    }

    public static String getToken(Context context) {
        SharedPreferences sharedPref = context.getSharedPreferences(context.getString(R.string.preference_file_key), Context.MODE_PRIVATE);
        return sharedPref.getString(context.getString(R.string.saved_token), null);
    }
}

File: app/src/main/java/com/coopra/nebulus/TokenHandler.kt
package com.coopra.nebulus

import android.content.Context
import androidx.core.content.edit
import androidx.preference.PreferenceManager

class TokenHandler {
    fun saveToken(context: Context, token: String) {
        PreferenceManager.getDefaultSharedPreferences(context).edit(commit = true) {
            putString(TOKEN_KEY, token)
        }
    }

    fun getToken(context: Context): String? {
        return PreferenceManager.getDefaultSharedPreferences(context).getString(TOKEN_KEY, null)
    }

    companion object {
        private const val TOKEN_KEY = "saved_token"
    }
}